features/ui/hero/hero.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect, useState } from "react";
+import Head from "next/head";
+import styles from "./hero.module.scss";
+
+// Define the types for your data structure
+interface Image {
+  src: string;
+  width: number;
+  height: number;
+}
+
+interface HeroSection {
+  sectionType: "hero";
+  theme: string;
+  title: string;
+  subtitle: string;
+  image: Image;
+}
+
+interface ContentData {
+  meta: {
+    title: string;
+    description: string;
+    image: string;
+  };
+  sections: HeroSection[];
+}
+
+export function Hero() {
+  const [heroData, setHeroData] = useState<HeroSection | null>(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const res = await fetch("https://prolog-api.profy.dev/content-page/home");
+      const result: ContentData = await res.json();
+      const heroSection = result.sections.find(
+        (section): section is HeroSection => section.sectionType === "hero",
+      );
+      setHeroData(heroSection || null);
+    };
+
+    fetchData();
+  }, []);
```

Comment:
I see a few problems with this:

1. The project uses react query to manage API requests. It would be more consistent to use rc here as well.
2. Actually I wouldn't call the API at this level. This is just one section of the whole page. Imagine you add the other sections mentioned in the API data as well. You wouldn't fetch the same data over and over again. 

Thus it makes more sense to move the API call to the page level (meaning the page component file). Then you can iterate over the sections in the data and render the appropriate component depending on the `sectionType` field.

As another improvement I would move this API call to the server side. With this version of Next.js you can use the `getStaticProps` function in the page component (double check if that's correct as I'm typing this comment from memory). Then the pages are pre-fetched and built during build time making their delivery much faster. That in turn is good for SEO